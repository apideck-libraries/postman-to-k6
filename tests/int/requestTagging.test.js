/* eslint-disable space-before-function-paren */

import convertFile from '../../lib/convert/file';

test('request --k6-request-tagging=request', async () => {
  const options = {
    k6RequestTagging: 'request',
  };
  const [main] = await convertFile('test/material/2/request.json', options);
  expect(main).toMatchSnapshot();
});

test('request --k6-request-tagging=folder-request', async () => {
  const options = {
    k6RequestTagging: 'folder-request',
  };
  const [main] = await convertFile('test/material/2/request.json', options);
  expect(main).toMatchSnapshot();
});

test('request no request-tagging', async () => {
  const options = {
    k6RequestTagging: '',
  };
  const [main] = await convertFile('test/material/2/request.json', options);
  expect(main).toMatchSnapshot();
});

test('folder request --k6-request-tagging=request', async () => {
  const options = {
    k6RequestTagging: 'request',
  };
  const [main] = await convertFile(
    'test/material/2/inherit-folder.json',
    options
  );
  expect(main).toMatchSnapshot();
});

test('folder request --k6-request-tagging=folder-request', async () => {
  const options = {
    k6RequestTagging: 'folder-request',
  };
  const [main] = await convertFile(
    'test/material/2/inherit-folder.json',
    options
  );
  expect(main).toMatchSnapshot();
});

test('folder request no request-tagging', async () => {
  const options = {
    k6RequestTagging: '',
  };
  const [main] = await convertFile(
    'test/material/2/inherit-folder.json',
    options
  );
  expect(main).toMatchSnapshot();
});

test('request separate --k6-request-tagging=request', async () => {
  const options = {
    separate: true,
    k6RequestTagging: 'request',
  };
  const [main, requests] = await convertFile(
    'test/material/2/request.json',
    options
  );
  // remove ID from requests
  let lines = requests['TestRequest.js'].split('\n');
  lines.splice(2, 1);
  requests['TestRequest.js'] = lines.join('\n');
  expect(requests).toMatchSnapshot();
});

test('request separate --k6-request-tagging=folder-request', async () => {
  const options = {
    separate: true,
    k6RequestTagging: 'folder-request',
  };
  const [main, requests] = await convertFile(
    'test/material/2/request.json',
    options
  );
  // remove ID from requests
  let lines = requests['TestRequest.js'].split('\n');
  lines.splice(2, 1);
  requests['TestRequest.js'] = lines.join('\n');
  expect(requests).toMatchSnapshot();
});

test('request separate no request-tagging', async () => {
  const options = {
    separate: true,
    k6RequestTagging: '',
  };
  const [main, requests] = await convertFile(
    'test/material/2/request.json',
    options
  );
  // remove ID from requests
  let lines = requests['TestRequest.js'].split('\n');
  lines.splice(2, 1);
  requests['TestRequest.js'] = lines.join('\n');
  expect(requests).toMatchSnapshot();
});

test('folder request separate --k6-request-tagging=request', async () => {
  const options = {
    separate: true,
    k6RequestTagging: 'request',
  };
  const [main, requests] = await convertFile(
    'test/material/2/inherit-folder.json',
    options
  );
  // remove ID from requests
  let lines = requests['TestFolder']['TestRequest.js'].split('\n');
  lines.splice(2, 1);
  requests['TestFolder']['TestRequest.js'] = lines.join('\n');
  expect(requests).toMatchSnapshot();
});

test('folder request separate --k6-request-tagging=folder-request', async () => {
  const options = {
    separate: true,
    k6RequestTagging: 'folder-request',
  };
  const [main, requests] = await convertFile(
    'test/material/2/inherit-folder.json',
    options
  );
  // remove ID from requests
  let lines = requests['TestFolder']['TestRequest.js'].split('\n');
  lines.splice(2, 1);
  requests['TestFolder']['TestRequest.js'] = lines.join('\n');
  expect(requests).toMatchSnapshot();
});

test('folder request separate no request-tagging', async () => {
  const options = {
    separate: true,
    k6RequestTagging: '',
  };
  const [main, requests] = await convertFile(
    'test/material/2/inherit-folder.json',
    options
  );
  // remove ID from requests
  let lines = requests['TestFolder']['TestRequest.js'].split('\n');
  lines.splice(2, 1);
  requests['TestFolder']['TestRequest.js'] = lines.join('\n');
  expect(requests).toMatchSnapshot();
});
