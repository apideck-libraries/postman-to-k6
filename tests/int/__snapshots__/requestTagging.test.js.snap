// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`folder request --k6-request-tagging=folder-request 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import { group } from "k6";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  group("TestFolder", function() {
    postman[Request]({
      name: "TestRequest",
      method: "GET",
      address: "http://example.com/",
      tags: { name: "TestFolder - TestRequest" },
      auth(config, Var) {
        config.headers.Authorization = "Bearer token";
      }
    });
  });
}
"
`;

exports[`folder request --k6-request-tagging=request 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import { group } from "k6";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  group("TestFolder", function() {
    postman[Request]({
      name: "TestRequest",
      method: "GET",
      address: "http://example.com/",
      tags: { name: "TestRequest" },
      auth(config, Var) {
        config.headers.Authorization = "Bearer token";
      }
    });
  });
}
"
`;

exports[`folder request no request-tagging 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import { group } from "k6";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  group("TestFolder", function() {
    postman[Request]({
      name: "TestRequest",
      method: "GET",
      address: "http://example.com/",
      auth(config, Var) {
        config.headers.Authorization = "Bearer token";
      }
    });
  });
}
"
`;

exports[`folder request separate --k6-request-tagging=folder-request 1`] = `
{
  "TestFolder": {
    "TestRequest.js": "postman[Symbol.for("define")]({
  name: "TestRequest",
  method: "GET",
  address: "http://example.com/",
  tags: { name: "TestFolder - TestRequest" },
  auth(config, Var) {
    config.headers.Authorization = "Bearer token";
  }
});
",
  },
}
`;

exports[`folder request separate --k6-request-tagging=request 1`] = `
{
  "TestFolder": {
    "TestRequest.js": "postman[Symbol.for("define")]({
  name: "TestRequest",
  method: "GET",
  address: "http://example.com/",
  tags: { name: "TestRequest" },
  auth(config, Var) {
    config.headers.Authorization = "Bearer token";
  }
});
",
  },
}
`;

exports[`folder request separate no request-tagging 1`] = `
{
  "TestFolder": {
    "TestRequest.js": "postman[Symbol.for("define")]({
  name: "TestRequest",
  method: "GET",
  address: "http://example.com/",
  auth(config, Var) {
    config.headers.Authorization = "Bearer token";
  }
});
",
  },
}
`;

exports[`request --k6-request-tagging=folder-request 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/",
    tags: { name: "null - TestRequest" }
  });
}
"
`;

exports[`request --k6-request-tagging=request 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/",
    tags: { name: "TestRequest" }
  });
}
"
`;

exports[`request no request-tagging 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/"
  });
}
"
`;

exports[`request separate --k6-request-tagging=folder-request 1`] = `
{
  "TestRequest.js": "postman[Symbol.for("define")]({
  name: "TestRequest",
  method: "GET",
  address: "http://example.com/",
  tags: { name: "null - TestRequest" }
});
",
}
`;

exports[`request separate --k6-request-tagging=request 1`] = `
{
  "TestRequest.js": "postman[Symbol.for("define")]({
  name: "TestRequest",
  method: "GET",
  address: "http://example.com/",
  tags: { name: "TestRequest" }
});
",
}
`;

exports[`request separate no request-tagging 1`] = `
{
  "TestRequest.js": "postman[Symbol.for("define")]({
  name: "TestRequest",
  method: "GET",
  address: "http://example.com/"
});
",
}
`;
